frontend/tests/artillery/processor.ts

```diff
@@ -0,0 +1,123 @@
+import { readFile } from "fs/promises";
+import { random } from "lodash";
+
+type dataType = {
+  ids: {
+    [key: string]: Array<number>;
+  };
+  queries: Array<string>;
+  pages: Array<string>;
+  status: Array<string>;
+  agencies: Array<string>;
+  funding: Array<string>;
+  eligibility: Array<string>;
+  category: Array<string>;
+};
+type globalVars = {
+  $environment?: string;
+};
+
+type returnVars = {
+  id: number;
+  query: string;
+  route: string;
+  pages: string;
+};
+
+// eslint-disable-next-line @typescript-eslint/require-await
+async function getOppId(context: { vars: dataType & returnVars & globalVars }) {
+  const env = context.vars.$environment as string;
+  context.vars.id =
+    context.vars.ids[env][random(context.vars.ids[env].length - 1)];
+}
+
+// eslint-disable-next-line @typescript-eslint/require-await
+async function get404(context: { vars: returnVars }) {
+  const num = random(10);
+  // ~50% of 404s are opp pages.
+  if (num % 2 !== 0) {
+    context.vars.route = `opportunity/${num}`;
+  } else {
+    context.vars.route = randomString(num);
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/require-await
+async function getStatic(context: { vars: returnVars }) {
+  context.vars.route =
+    context.vars.pages[random(context.vars.pages.length - 1)];
+}
+
+// eslint-disable-next-line @typescript-eslint/require-await
+async function getSearchQuery(context: { vars: returnVars & dataType }) {
+  const { queries, status, agencies, eligibility, category } = context.vars;
+  const qu = `query=${queries[random(queries.length - 1)]}`;
+  const st = `status=${status[random(status.length - 1)]}`;
+  const ag = `agency=${agencies[random(agencies.length - 1)]}`;
+  const cat = `category=${category[random(category.length - 1)]}`;
+  const el = `eligibility=${eligibility[random(eligibility.length - 1)]}`;
+  const pager = `page=${random(5)}`;
+  // Most search params only include the queries, but smaller percent include
+  // filters. This allows configuring that percent for composing the query
+  const weights = [
+    {
+      percent: 50,
+      params: [qu, st, pager],
+    },
```

Comment:
`qu, st, pager` what do these mean?

Reply:
I put a little more explanation into the desc. This composes the query so that 50% of the requests use a query term, pager, and status selected...

Reply:
Oh I see it now, can we use the full name though pls? rg. `eligibility`

Reply:
I updated it to be more verbose.